master/internal/rm/kubernetesrm/pods.go

```diff
@@ -917,26 +917,25 @@ func (p *pods) cleanUpPodHandler(ctx *actor.Context, podHandler *actor.Ref) erro
 func (p *pods) handleAPIRequest(ctx *actor.Context, apiCtx echo.Context) {
 	switch apiCtx.Request().Method {
 	case echo.GET:
-		summary, err := p.summarize(ctx)
-		if err != nil {
-			ctx.Respond(apiCtx.JSON(http.StatusInternalServerError, err))
+		summaries := p.summarizeClusterByNodes(ctx)
+		_, nodesToPools := p.getNodeResourcePoolMapping(summaries)
+		for nodeName, summary := range summaries {
+			summary.ResourcePool = nodesToPools[summary.ID]
+			summaries[nodeName] = summary
 		}
-		ctx.Respond(apiCtx.JSON(http.StatusOK, summary))
+		ctx.Respond(apiCtx.JSON(http.StatusOK, summaries))
```

Comment:
can we move this logic from here and 933-934 into `p.summarizeClusterByNodesWithPools` or something?

Reply:
alternatively, just have `summarizeClusterByNodes` do the adding resource pool to agent summaries bit